.circleci/config.yml

```diff
@@ -2062,16 +2061,93 @@ jobs:
       - run:
           name: Build kona host
           command: just build-kona-host
-          working_directory: kona
+          working_directory: kona-proofs
       - save_cache:
-          key: kona-host-{{ checksum "./kona/justfile" }}-{{ checksum "./kona/version.json" }}
+          key: kona-host-{{ checksum "./kona-proofs/justfile" }}-{{ checksum "./kona-proofs/version.json" }}
           name: Save Kona host to cache
           paths:
-            - "kona/bin/kona-host"
+            - "kona-proofs/bin/kona-host"
+      - persist_to_workspace:
+          root: .
+          paths:
+            - "kona-proofs/bin/kona-host"
+
+  # Build a single Rust binary for sysgo tests (parameterized).
+  rust-binary-build:
+    docker:
+      - image: <<pipeline.parameters.default_docker_image>>
+    resource_class: xlarge
+    parameters:
+      cache_name:
+        description: "Identifier for cache key and logging"
+        type: string
+      directory:
+        description: "Directory containing the Cargo workspace"
+        type: string
+      build_command:
+        description: "Cargo build command to run"
+        type: string
+      check_binary:
+        description: "Binary name to check for cache hit (in target/debug/)"
+        type: string
+    steps:
+      - utils/checkout-with-mise:
+          checkout-method: blobless
+      - run:
+          name: Initialize << parameters.directory >> submodule (shallow)
+          command: git submodule update --init --recursive --depth 1 << parameters.directory >>
+      - restore_cache:
+          name: Restore << parameters.cache_name >> cache
+          keys:
+            - rust-<< parameters.cache_name >>-v1-{{ checksum "<< parameters.directory >>/Cargo.lock" }}
+            - rust-<< parameters.cache_name >>-v1-
+      - run:
+          name: Check cache status
+          command: |
+            if [ -f "<< parameters.directory >>/target/debug/<< parameters.check_binary >>" ]; then
+              echo "export CACHE_HIT=true" >> "$BASH_ENV"
+              echo "Cache hit - << parameters.check_binary >> exists"
+            else
+              echo "export CACHE_HIT=false" >> "$BASH_ENV"
+              echo "Cache miss - will build"
+            fi
+      - run:
+          name: Install clang (if cache miss)
```

Comment:
Could enable this step only optionally for `op-rbuilder` with a bool param to `rust-binary-build`